static/app/views/starfish/views/spanSummaryPage/spanSummaryView.tsx

```diff
@@ -0,0 +1,257 @@
+  getThroughputChartTitle,
+  getThroughputTitle,
+} from 'sentry/views/starfish/views/spans/types';
+import {Block, BlockContainer} from 'sentry/views/starfish/views/spanSummaryPage/block';
+import {getSpanOperationDescription} from 'sentry/views/starfish/views/spanSummaryPage/getSpanOperationDescription';
+
+const CHART_HEIGHT = 160;
+
+interface Props {
+  groupId: string;
+}
+
+type Query = {
+  endpoint: string;
+  endpointMethod: string;
+  transaction: string;
+  transactionMethod: string;
+};
+
+export function SpanSummaryView({groupId}: Props) {
+  const location = useLocation<Query>();
+  const {endpoint, endpointMethod} = location.query;
+
+  const {data: fullSpan} = useFullSpanFromTrace(groupId);
+
+  const queryFilter: SpanSummaryQueryFilters = endpoint
+    ? {transactionName: endpoint, 'transaction.method': endpointMethod}
+    : {};
+
+  if (endpointMethod && queryFilter) {
+    queryFilter['transaction.method'] = endpointMethod;
+  }
+
+  const {data: spanMetrics} = useSpanMetrics(
+    groupId,
+    queryFilter,
+    [
+      SpanMetricsFields.SPAN_OP,
+      SpanMetricsFields.SPAN_DESCRIPTION,
+      SpanMetricsFields.SPAN_ACTION,
+      SpanMetricsFields.SPAN_DOMAIN,
+      'count()',
+      'spm()',
+      `sum(${SpanMetricsFields.SPAN_SELF_TIME})`,
+      `avg(${SpanMetricsFields.SPAN_SELF_TIME})`,
+      'time_spent_percentage()',
+      'http_error_count()',
+    ],
+    'api.starfish.span-summary-page-metrics'
+  );
+
+  const span = {
+    ...spanMetrics,
+    [SpanMetricsFields.SPAN_GROUP]: groupId,
+  } as {
+    [SpanMetricsFields.SPAN_OP]: string;
+    [SpanMetricsFields.SPAN_DESCRIPTION]: string;
+    [SpanMetricsFields.SPAN_ACTION]: string;
+    [SpanMetricsFields.SPAN_DOMAIN]: string;
+    [SpanMetricsFields.SPAN_GROUP]: string;
+  };
+
+  const {isLoading: areSpanMetricsSeriesLoading, data: spanMetricsSeriesData} =
+    useSpanMetricsSeries(
+      groupId,
+      queryFilter,
+      [`avg(${SpanMetricsFields.SPAN_SELF_TIME})`, 'spm()', 'http_error_count()'],
+      'api.starfish.span-summary-page-metrics-chart'
+    );
+
+  useSynchronizeCharts([!areSpanMetricsSeriesLoading]);
+
+  const spanMetricsThroughputSeries = {
+    seriesName: span?.[SpanMetricsFields.SPAN_OP]?.startsWith('db')
+      ? 'Queries'
+      : 'Requests',
+    data: spanMetricsSeriesData?.['spm()'].data,
+  };
+
+  const spanOperationDescription = getSpanOperationDescription(
+    span[SpanMetricsFields.SPAN_OP]
+  );
+
+  return (
+    <React.Fragment>
+      <HeaderContainer>
+        <FilterOptionsContainer>
+          <StarfishDatePicker />
+        </FilterOptionsContainer>
+
+        <BlockContainer>
+          {span?.[SpanMetricsFields.SPAN_OP]?.startsWith('db') &&
+            span?.[SpanMetricsFields.SPAN_OP] !== 'db.redis' && (
+              <Block title={t('Table')}>{span?.[SpanMetricsFields.SPAN_DOMAIN]}</Block>
+            )}
+
+          <Block
+            title={getThroughputTitle(span?.[SpanMetricsFields.SPAN_OP])}
```

Comment:
feels weird to have all these optional chaining operators around here w/ span, but maybe we can fix this afterwards with a bigger types refactor.

Reply:
Agreed, though when `span` comes from the network, there are lot of quirky places where information might be partially unavailable. We should clarify this regardless, though